### DISCUSSION TITLE
Gmsh physical curve (line) names disappear for a 3D mesh but not a 2D mesh

### DISCUSSION THREAD

**Discussion by ScepticalRabbit** [idaholab/moose#27607] [Q&A Meshing] [answered]
I have built a 3D mesh in Gmsh to run a thermo-mechanical analysis in MOOSE (see case09 *.i and *.geo files here: https://github.com/Applied-Materials-Technology/pyvale/tree/lf-dev/simcases/case09). I can define physical surfaces and points which appear with the correct name in the set list if I run it through MOOSE and open the exodus in paraview. However, the physical curves (lines) do not appear with the name I give them in the set list after running the thermal analysis but they do appear in gmsh with the correct names. Looking through the node sets in paraview it appears that the physical curves I have defined are appearing as 'nodeset_5' and 'nodeset_6' but they do not have the name specified in gmsh so I cannot apply any boundary conditions on lines in 3D using the specified name.

Interestingly, if I run a 2D gmsh file and define a physical curve everything works fine and I can define boundary conditions using the name I specified for the physical curve in the gmsh file (case07 *.i and *.geo files here: https://github.com/Applied-Materials-Technology/pyvale/tree/lf-dev/simcases/case07).

I can get around this by applying my mechanical boundary conditions on a series of physical points but it should be possible to apply boundary conditions on lines (physical curves) for a 3D mesh. Am I missing something in specifying the gmsh physical curves in 3D or is this a quirk of passing a gmsh mesh to MOOSE? 

![case07](https://github.com/idaholab/moose/assets/121958155/8d2627e9-04a9-4b4c-9637-4ddbe5648fb4)
![case09](https://github.com/idaholab/moose/assets/121958155/689ab722-0b49-4fbe-8a76-62effcc82b38)

**Comment by GiudGiud** [comment, marked as answer]
Hello

Not passing the name seems like a quirk of the gmsh mesh-reading code in libMesh.
If you feel so inclined, it could be changed there.
If not, you can use this to re-establish the sideset names:
https://mooseframework.inl.gov/source/meshgenerators/RenameBoundaryGenerator.html

Guillaume
